src/main/java/com/icthh/xm/ms/timeline/config/TopicManagerConfiguration.java

```diff
@@ -0,0 +1,49 @@
+package com.icthh.xm.ms.timeline.config;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.icthh.xm.commons.logging.trace.SleuthWrapper;
+import com.icthh.xm.commons.security.XmAuthenticationContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.commons.topic.TopicManager;
+import com.icthh.xm.commons.topic.message.LepMessageHandler;
+import com.icthh.xm.commons.topic.message.MessageHandler;
+import com.icthh.xm.commons.topic.message.MessageService;
+import com.icthh.xm.lep.api.LepManager;
+import com.icthh.xm.ms.timeline.service.DomainEventService;
+import com.icthh.xm.ms.timeline.service.kafka.DomainEventMessageHandler;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.core.KafkaTemplate;
+
+@Configuration
+@RequiredArgsConstructor
+public class TopicManagerConfiguration {
+
+    private final ObjectMapper objectMapper;
+    private final DomainEventService domainEventService;
+    private final MessageService messageService;
+    private final TenantContextHolder tenantContextHolder;
+    private final XmAuthenticationContextHolder authContextHolder;
+    private final LepManager lepManager;
+
+    @Bean
+    public TopicManager topicManager(@Value("${spring.application.name}") String appName,
+                                     KafkaProperties kafkaProperties,
+                                     KafkaTemplate<String, String> kafkaTemplate,
+                                     SleuthWrapper sleuthWrapper) {
+        return new TopicManager(appName, kafkaProperties, kafkaTemplate, consumerMessageHandler(), sleuthWrapper);
```

Comment:
how about use this one
https://github.com/xm-online/xm-commons/pull/182

Reply:
when it will be merged?